roles/nfs-server/tasks/configure_lvm.yml

```diff
@@ -0,0 +1,41 @@
+---
+
+- name: "Set VG name"
+  set_fact:
+    nfs_vg_name: "{{ nfs_vg_name | default(default_nfs_vg_name) }}"
+
+- name: "Set LV name"
+  set_fact:
+    nfs_lv_name: "{{ nfs_lv_name | default(default_nfs_lv_name) }}"
+
+- name: "Set NFS share basename"
+  set_fact:
+    nfs_share_basedir: "{{ nfs_share_basedir | default(default_nfs_share_basedir) }}"
+
+- name: "Setup PV"
+  shell: "pvcreate {{ nfs_storage_device }}" 
+  when: lvm_check.rc != 0
+
+- name: "Setup VG"
+  shell: "vgcreate {{ nfs_vg_name }} {{ nfs_storage_device }}"
+
+- name: "Setup LV"
```

Comment:
why not use lvol ansible module  http://docs.ansible.com/ansible/lvol_module.html